File: examples/php/rest.php
<?php
// Minimal REST example without external deps
// Run: php examples/php/rest.php

$url = getenv('FAUXMAIL_HTTP') ?: 'http://127.0.0.1:8025/send';
$payload = json_encode([
  'from' => 'dev@example.test',
  'to' => ['you@example.test'],
  'subject' => 'Hello via REST (PHP)',
  'text' => 'Hi from PHP using REST',
]);
$opts = [
  'http' => [
    'method' => 'POST',
    'header' => "Content-Type: application/json\r\n",
    'content' => $payload,
    'ignore_errors' => true,
  ]
];
$ctx = stream_context_create($opts);
$res = file_get_contents($url, false, $ctx);
echo $res, "\n";

File: examples/python/rest.py
import json, os, urllib.request

url = os.getenv('FAUXMAIL_HTTP', 'http://127.0.0.1:8025/send')
data = {
    'from': 'dev@example.test',
    'to': ['you@example.test'],
    'subject': 'Hello via REST (Python)',
    'text': 'Hi from Python using REST',
}
req = urllib.request.Request(url, data=json.dumps(data).encode('utf-8'), headers={'Content-Type': 'application/json'})
with urllib.request.urlopen(req) as resp:
    print(resp.read().decode('utf-8'))

